cli/azd/resources/scaffold/templates/resources.bicept

```diff
@@ -486,14 +485,14 @@ resource localUserOpenAIIdentity 'Microsoft.Authorization/roleAssignments@2022-0
 }
 {{- end}}
 
-{{- range .Services}}
+{{- range $service := .Services}}
 
 module {{bicepName .Name}}Identity 'br/public:avm/res/managed-identity/user-assigned-identity:0.2.1' = {
   name: '{{bicepName .Name}}identity'
   params: {
     name: '${abbrs.managedIdentityUserAssignedIdentities}{{bicepName .Name}}-${resourceToken}'
     location: location
-    {{- if (or (and .DbPostgres (eq .DbPostgres.AuthType "USER_ASSIGNED_MANAGED_IDENTITY")) (and .DbMySql (eq .DbMySql.AuthType "USER_ASSIGNED_MANAGED_IDENTITY")))}}
+    {{- if (or (and $.DbPostgres (eq $.DbPostgres.AuthType "USER_ASSIGNED_MANAGED_IDENTITY")) (and $.DbMySql (eq $.DbMySql.AuthType "USER_ASSIGNED_MANAGED_IDENTITY")))}}
```

Comment:
Add `$` to access variables in outer scope. Here, current scope in serviceSpec, and the outer scope is infraSpec.